content/client-lib-development-guide/features.textile

```diff
@@ -222,6 +222,10 @@ h3(#rest-channel). Channel
 *** @(RSL1g4)@ When publishing a message with an explicit @clientId@ that is incompatible with the identified client's @clientId@ (either inferred or explicitly configured in the token or @ClientOptions@), the library will reject the message immediately and indicate an error
 ** @(RSL1h)@ Where the library language permits, the @Channel#publish(name, data)@ method should provide an optional argument that allows the @clientId@ value to be specified such as @Channel#publish('event', 'data', { clientId: 'John' })@, and/or the @extras@ field to be specified such as @Channel#publish('event', 'data', { extras: { push: { 'key': 'value' } })@
 ** @(RSL1i)@ If the total size of the message or (if publishing an array) messages, calculated per "TO3l8":#TO3l8, exceeds the @maxMessageSize@, then the client library should reject the publish and indicate an error
+** @(RSL1j)@ When @Message@ objects are provided, any valid @Message@ attribute (that is, an attribute specified in "TM2":#TM2) that is supplied by the caller must be included in the encoded message. (This does not mean it must be included _unaltered_; for example the @data@ and @encoding@ will be subject to processing per "RSL4":#RSL4)
+*** @(RSL1j1)@ A test should exist which publishes a @Message@ which includes a fixed @id@ twice, and checks that it is received only once
+** @(RSL1k)@ If @idempotentRestPublishing@ (see "TO3n":#TO3n) is enabled, if the @Message@s do not include caller-supplied @ids@, for each @Message@ in the array the library should randomly generate get 9 bytes from a source of randomness, encode them as base64, and set it as the @Message@ @id@
+*** @(RSL1k1)@ For the purposes of deciding whether the @Message@s include caller-supplied @ids@, the library may assume that the first @Message@ does not have an @id@ then none of them do
```

Comment:
Why?

Reply:
see the bit in parens in last paragraph of original comment